### PR TITLE
Moved Kernel and KernelDb unitTests to GTest

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

Changed Kernel and KernelDb unit tests to Gtest

## Related Issue

Continued gtest conversion

## Motivation and Context

This gets the unit tests off the old file comparison based testing and to in-memory testing.

## How Has This Been Tested?

All tests compiled and passed locally on Ubuntu and Mac.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by Kelvinrr — DISMISSED** [review]

**Comment by Kelvinrr** [comment]
I'll leave it open a bit more for others to see it then merge it tonight for the nighties to run if no one has objections.

File: isis/src/system/objs/Kernel/Kernel.h [hunk 1]
```diff
@@ -87,5 +87,17 @@ namespace Isis {
...
+
+  /**
+   * Logical operator for combining Type flags.
+   *
+   * @param a The first Type flag.
+   * @param b The second Type flag.
+   *
+   * @return Type flag that contains all Types in a and all Types in b.
+   */
+  inline Kernel::Type operator|(Kernel::Type a, Kernel::Type b) {
```
Comment (scsides) [inline isis/src/system/objs/Kernel/Kernel.h]: My understanding is the "inline" is ignored by all modern compilers.

**Review by scsides — DISMISSED** [review]
Looks clean

File: isis/src/system/objs/Kernel/Kernel.h (hunk 1 above)
Comment (jessemapel) [reply isis/src/system/objs/Kernel/Kernel.h]: yeah it does nothing. I'm going to pull it to make it cleaner. Our style-guide actually disallows it without testing right now.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**